packages/angular/cli/utilities/package-metadata.ts

```diff
@@ -107,25 +109,48 @@ function readOptions(
     logger.info(`Locating potential ${baseFilename} files:`);
   }
 
-  let options: { [key: string]: string } = {};
+  const options: PackageManagerOptions = {};
   for (const location of [...defaultConfigLocations, ...projectConfigLocations]) {
     if (existsSync(location)) {
       if (showPotentials) {
         logger.info(`Trying '${location}'...found.`);
       }
 
       const data = readFileSync(location, 'utf8');
-      options = {
-        ...options,
-        ...(yarn ? lockfile.parse(data) : ini.parse(data)),
-      };
-
-      if (options.cafile) {
-        const cafile = path.resolve(path.dirname(location), options.cafile);
-        delete options.cafile;
-        try {
-          options.ca = readFileSync(cafile, 'utf8').replace(/\r?\n/, '\\n');
-        } catch {}
+      // Normalize RC options that are needed by 'npm-registry-fetch'.
+      // See: https://github.com/npm/npm-registry-fetch/blob/ebddbe78a5f67118c1f7af2e02c8a22bcaf9e850/index.js#L99-L126
+      const rcConfig: PackageManagerOptions = yarn ? lockfile.parse(data) : ini.parse(data);
+      for (const [key, value] of Object.entries(rcConfig)) {
+        switch (key) {
+          case 'noproxy':
+          case 'no-proxy':
+            options['noProxy'] = value;
+            break;
+          case 'maxsockets':
+            options['maxSockets'] = value;
+            break;
+          case 'https-proxy':
+          case 'proxy':
+            options['proxy'] = value;
+            break;
+          case 'strict-ssl':
+            options['strictSSL'] = value;
+            break;
+          case 'local-address':
+            options['localAddress'] = value;
+            break;
+          case 'cafile':
+            if (typeof value === 'string') {
+              const cafile = path.resolve(path.dirname(location), value);
+              try {
+                options['ca'] = readFileSync(cafile, 'utf8').replace(/\r?\n/, '\\n');
```

Comment:
Do you know why is removing `\r` necessary?
Also, do we need the global `g` flag for the regex? The old code doesn't use the flag, but it seems like a bug.

Reply:
I’ll investigate.

Reply:
Latest npm code for `cafile`/`ca` option: https://github.com/npm/config/blob/95d566201587931dccb71ef5ca0df826f440e191/lib/index.js#L699-L702